apollo-runtime/src/main/java/com/apollographql/apollo/ApolloCall.java

```diff
@@ -25,10 +27,22 @@
 
   @Nonnull ApolloCall<T> clone();
 
-  interface Callback<T> {
-    void onResponse(@Nonnull Response<T> response);
+  abstract class Callback<T> {
+    public abstract void onResponse(@Nonnull Response<T> response);
+
+    public abstract void onFailure(@Nonnull ApolloException e);
+
+    public void onHttpError(@Nonnull ApolloHttpException e) {
+      onFailure(e);
```

Comment:
Thoughts on calling `closeQuietly(e.response())` after we call `onFailure`? I think it would prevent a lot of leaked connections